### PR TITLE
Food Cart remade as a mini-kitchen you can set up in the hallways (AND DO ANYTHING) (ARMFLOYDENKITCHENEZAB) 

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

The food cart no longer... Does whatever garbage it did before. What, like, store condiments? Terrible. Now you drag it around and set it up, creating a tent with a griddle, smartfridge, and a table. You can pack up anytime. Breaking parts of the tent will pack up the machine and subsequently break it, so careful!

## Why It's Good For The Game

The food cart has been an entirely outdated, out classed, completely dead machine to use. We just can do a lot better in 2020. Now it once again has a nice unique function and lets chefs do a new gimmick of being a travelling seller of food. Employs the "steal the janicart" meta from the janitor in the new and equally infuriating "steal the entire damn tent lmao"

## Changelog
:cl:
add: Food Cart has been completely remade to work as a mobile kitchen
/:cl:

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
can you make food carts order able from cargo?

**Comment by tralezab** [comment]
Not against it

**Comment by tralezab** [comment]
Also, here's what your setup looks like, dog.

**Comment by tralezab** [comment]
(I already custom did the tent and the griddle, maybe i should just say fuck it and do the table and fridge too)

**Comment by MadMedicineMan** [comment]
That's an amazing idea, I really like it and look forward to trying it :D

**Comment by Mickyan** [comment]
This is awesome, I really wish we had something like this for the bartender too

**Comment by Farquaar** [comment]
Can I still store food in it?

File: code/modules/food_and_drinks/kitchen_machinery/griddle.dm [hunk 1]
```diff
@@ -108,3 +111,11 @@
...
 	icon_state = "griddle[variant]_[on ? "on" : "off"]"
 
+/obj/machinery/griddle/stand
+	name = "griddle stand"
+	desc = "A more commercialized version of your traditional griddle. What happened to the good old days where people griddled with passion?"
+	variant = "stand"
+
+/obj/machinery/griddle/stand/Initialize()
+	. = ..()
+	add_overlay("front_bar")
```
Comment (ShizCalev) [inline code/modules/food_and_drinks/kitchen_machinery/griddle.dm:121]: this should be in update_overlays()

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by tralezab** [comment]
@Farquaar 

<details>
  <summary>spoiler</summary>
  
  image
  
</details>

**Comment by Farquaar** [comment]
> 
> 
> @Farquaar
> i think you are a

Is it possible for you to make the new expandable snack cart optional? i.e. the snack cart has its original functionality by default, but there's a button on the interface that allows you to expand it, transferring its contents to the smartfridge?

**Comment by tralezab** [comment]
Listen. It's glass dispensing feature starts out with you requiring to fill it with glass. You add your own found reagents I assume from working with the bartender, and then you finally go out and find that the food cart has been actually broken for 2-6 years and refuses to dispense reagents, making it's only usable feature working as a shittier smart fridge.

Is this /really/ something people will miss? How do you not report a massive bug like half the machine not working for years on end if this is something people use in any capacity

**Comment by Farquaar** [comment]
Not sure what I can say to convince you. All I'm telling you is that a 1x1 portable food dispenser can be used in ways that a 3x2 multi-tile machine cannot. 1x1 is a convenient, compact way to distribute food without leaving it in haphazard heaps on your front counter.

I would much rather that this expansion feature be an addition, rather than a replacement for the food cart's base functionality.

**Comment by Tlaltecuhtli** [comment]
a
instead of large ugly fridge why not a smaller sprite version like in this photoshop i made so u have a bigger table? you might also fit a mini microwave with small sprite this way

**Comment by tralezab** [comment]
Sure, I'll try it.

**Comment by RaveRadbury** [comment]
This looks really unwieldly. I'm already used to setting up or finding a table. Please keep the old functions.

**Comment by LemonInTheDark** [comment]
I'll be honest with you, I don't trust the multitile nature to not be buggy as hell, or awkward as fuck, are these concerns irrational?

**Comment by tralezab** [comment]
Maybe but it can't break the game in any crazy way. The most you'll see is some funny guy blocking the hallways with it in which case it's easy to make permanently inoperable or simple one table to hop over

**Pull request merged**